pipeline/seed_data.py

```diff
@@ -0,0 +1,133 @@
+        region = plant['Region']
+        locations.append({"region": region, "country": country, "continent": continent})
+
+    location_sets = set([tuple(location.items()) for location in locations])
+    unique_locations = [dict(i) for i in location_sets]
+    return unique_locations
+
+
+def extract_plant_data(csv_data: list[dict],botanist_data: list[dict], location_data: list[dict]) -> list[dict]:
+    """Extracts relevant plant data from the cleaned .csv file, combines with the 
+    botanist and location data, returns a list of unique dicts to seed the plant table."""
+    plants = []
+    for plant in csv_data:
+        plant_id = plant["Id"]
+        name = plant['Name']
+        for botanist in botanist_data:
+            if plant['Botanist Name']== botanist['name']:
+                botanist_num = botanist['telephone_number']
+        for location in location_data:
+            if plant['Region'] == location['region']:
+                location_reg = location['region']
+
+        plants.append({"plant_id": plant_id, "name": name, "botanist_num": botanist_num, "location_reg": location_reg})
+    return plants
+
+
+def seed_botanist_table(conn: Connection, botanist_data: list[dict]) -> None:
+    """Seed the botanist table with the botanist data list."""
+    for botanist in botanist_data:
+        query = sql.text("INSERT INTO s_delta.botanist (name, email, telephone_number) VALUES (:n, :e, :t)")
+        args = ({"n":botanist["name"], "e":botanist["email"], "t":botanist["telephone_number"]})
+        conn.execute(query,args)
+
+
+
+def seed_location_table(conn: Connection, location_data: list[dict]) -> None:
+    """Seed the location table with the location data list."""
+    for location in location_data:
+        query = sql.text(
+            "INSERT INTO s_delta.location (region, country, continent) VALUES (:reg, :cou, :con)")
+        args = ({"reg": location["region"], "cou": location["country"],
+                "con": location["continent"]})
+        conn.execute(query, args)
+
+
+
+def seed_plant_table(conn: Connection, plant_data: list[dict]) -> None:
+    """Seed the plant table with the plant data list and relevant botanist and location ids."""
+    for plant in plant_data:
+        query = sql.text("SELECT botanist_id FROM s_delta.botanist WHERE telephone_number = (:num)")
+        args = ({"num": plant["botanist_num"]})
+        botanist_id = conn.execute(query,args).fetchone()[0]
+
+        query = sql.text(
+            "SELECT location_id FROM s_delta.location WHERE region = (:reg)")
+        args = ({"reg": plant["location_reg"]})
+        location_id = conn.execute(query, args).fetchone()[0]
+
+        query = sql.text(
+            "INSERT INTO s_delta.plant (plant_id, name, botanist_id, location_id) VALUES (:id, :n, :b_id, :l_id)")
+        args = ({"id": plant["plant_id"],"n": plant["name"], "b_id": botanist_id,
+                "l_id": location_id})
+        conn.execute(query, args)
```

Comment:
I like the seperation of querys and data throughout, its makes the code much more readable